compiler/souschef/include/souschef/Data/Constant.h

```diff
@@ -62,6 +64,90 @@ template <typename T> struct ConstantDataChefFactory : public DataChefFactory
   }
 };
 
+class ConstantInt4DataChef final : public DataChef
+{
+public:
+  ConstantInt4DataChef(const int8_t &value) : _value{value}
+  {
+    // DO NOTHING
+  }
+
+public:
+  std::vector<uint8_t> generate(int32_t count) const override
```

Comment:
Users of this function will not know how to interpret the returned vector. Could you add some comments?

```suggestion
  // Int4 constant is saved as int8 (extra 4 bits are filled with sign bits).
  // Callers must cast each element to int8 before using it.
  // Example
  // ConstInt4DataChef chef(-5)
  // auto values = chef.generate(3);
  // for (uint8_t value: values)
  //   int8_t real_value = static_cast<int8_t>(value);
  //   assert(value == 251 and real_value == -5);
  std::vector<uint8_t> generate(int32_t count) const override
```